sha2_256.c

```diff
@@ -143,6 +143,13 @@ void sha2_256_compress(void *v)
 /* initialize */
 
 static void sha2_256_init_h0(sha2_256_t *sha, const uint8_t *h0)
```

Comment:
While we are at it maybe we can add the lengths for fixed sized inputs. 
```suggestion
static void sha2_256_init_h0(sha2_256_t *sha, const uint8_t h0[32])
```